src/uct/cuda/cuda_ipc/cuda_ipc_cache.h

```diff
@@ -37,15 +67,21 @@ struct uct_cuda_ipc_cache {
 };
 
 
+typedef struct uct_cuda_ipc_remote_cache {
+    khash_t(cuda_ipc_rem_cache) hash;
+    ucs_recursive_spinlock_t    lock;
```

Comment:
do we need a lock both in "struct uct_cuda_ipc_cache" and also in here?
maybe can remove lock from uct_cuda_ipc_cache ?

Reply:
The attempt is to ensure that 
1. no two threads try to update a single rcache instance using uct_cuda_ipc_cache
2. no two threads try to update the hashtable for storing different rcache instances (one per per GPU per remote process) using remote_cache.lock. 

These two operations are independent so grabbing the locks for each should be independent too. I.e if a single thread decides to grab rache x for peer gpu x, then another thread should be allowed to grab rache y for peer gpu y. But two threads shouldn't be allowed to access rcache x simultaneously (at least for now we don't distinguish accessing rcache for read-only and for reads/writes)

Reply:
@yosefe when possible can we revisit this? I've addressed concern around ipc_map_lock here https://github.com/openucx/ucx/pull/5815#discussion_r565547712